.github/workflows/e3sm-gh-pages.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and deploy gh-pages branch with Mkdocs
+
+on:
+  # Runs every time master branch is updated
+  push:
+    branches: ["master"]
+  # Runs every time a PR is open against master
+  pull_request:
+    branches: ["master"]
+  workflow_dispatch:
+
+concurrency:
+  # Prevent 2+ copies of this workflow from running concurrently
+  group: e3sm-docs-action
+  
+jobs:
+  Build-and-Deploy-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          show-progress: false
+          fetch-depth: 0 # Needed, or else gh-pages won't be fetched, and push rejected
```

Comment:
At some point we may want to use another action for deployment. E.g., in scream repo we use `JamesIves/github-pages-deploy-action@v4`, which plays nicely with `rossjrw/pr-preview-action@v1`, which we use to deploy preview of PR docs.

As it is now, this action only builds the docs for a PR, but does not show it anywhere. We can merge as is, but I would soon work on making a preview of docs available on PR page.

Reply:
Yes lets do that in a later PR.